src/textual/widgets/_progress_bar.py

```diff
@@ -116,8 +160,28 @@ def render_indeterminate(self) -> RenderResult:
             highlight_range=(max(0, start), min(end, width)),
             highlight_style=Style.from_color(bar_style.color),
             background_style=Style.from_color(bar_style.bgcolor),
+            thickness=self.thickness,
         )
 
+    def _get_bar_style(self):
+        if self.colorscheme == "default":
+            return (
+                self.get_component_rich_style("bar--bar")
+                if self._percentage < 1
+                else self.get_component_rich_style("bar--complete")
+            )
+        elif self.colorscheme == "rainbow":
+            from_color = self.get_component_rich_style("bar--bar").color
+            target_color = self.get_component_rich_style("bar--complete").color
+            clockwise = False
+            factor = self._percentage if clockwise else -1.0 * self._percentage
+            bar_color = (
+                Color.from_rich_color(from_color)
+                .hsl_blend(Color.from_rich_color(target_color), factor)
+                .rich_color
+            )
```

Comment:
.hsl_blend(Color.from_rich_color(target_color), -1.0 * self._percentage)
I want to keep counter clockwise direction by default (because this is the longer path from blue to red, from cold color as low value till hot color as high value).

Reply:
Of course, you are absolutely right. I don't know why I thought the correct value for the factor was `1` 🤦